laalaa/apps/advisers/geocoder.py

```diff
@@ -37,7 +38,14 @@ def result_to_postcode(result):
 
 def lookup_postcode(postcode):
     normalised_postcode = normalise_postcode(postcode)
-    raw = requests.get(
+
+    session = requests.Session()
+
+    # `max_retries` only applies to failed DNS lookups, socket connections and connection timeouts, never to requests where data has made it to the server.
+    # by default, Requests does not retry failed connections. Please see documentation: https://docs.python-requests.org/en/latest/api/#requests.adapters.HTTPAdapter
+    session.mount("https://api.postcodes.io/", HTTPAdapter(max_retries=5))
+
```

Comment:
The code I found in the requests documentation was: 

s = Session()
retries = Retry(
    total=3,
    backoff_factor=0.1,
    status_forcelist=[502, 503, 504],
    allowed_methods={'POST'},
)
s.mount('https://', HTTPAdapter(max_retries=retries))

which looks a bit different from yours. Backoff is probably a good idea, especially for DNS (name) lookup failures as the failures themselves are often cached